SIT.Manager/ViewModels/LinuxSettingsPageViewModel.cs

```diff
@@ -0,0 +1,114 @@
+using Avalonia.Platform.Storage;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using SIT.Manager.Interfaces;
+using SIT.Manager.ManagedProcess;
+using SIT.Manager.Models;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace SIT.Manager.ViewModels;
+
+public partial class LinuxSettingsPageViewModel : ObservableObject
+{
+    // TODO: Check which services are needed for this ViewModel.
+    private readonly IManagerConfigService _configsService;
+    private readonly IPickerDialogService _pickerDialogService;
+    private readonly IBarNotificationService _barNotificationService;
+    private readonly ILocalizationService _localizationService;
+    
+    [ObservableProperty]
+    private LinuxConfig _config;
+    // DXVK Versions TODO
+    [ObservableProperty] 
+    private List<string> _dxvkVersions;
+    
+    public IAsyncRelayCommand ChangePrefixLocationCommand { get; }
+    public IAsyncRelayCommand ChangeRunnerLocationCommand { get; }
+    public IRelayCommand AddEnvCommand { get; }
+    public IRelayCommand DeleteEnvCommand { get; }
+
+    public LinuxSettingsPageViewModel(IManagerConfigService configService,
+                                      IPickerDialogService pickerDialogService,
+                                      IBarNotificationService barNotificationService,
+                                      ILocalizationService localizationService)
+    {
+        _configsService = configService;
+        _pickerDialogService = pickerDialogService;
+        _barNotificationService = barNotificationService;
+        _localizationService = localizationService;
+        
+        _config = (LinuxConfig) _configsService.Config;
+        
+        _config.PropertyChanged += (o, e) => OnPropertyChanged(e);
+        
+        // Find dxvk versions
+        //_dxvkVersions = _versionService.GetDXVKVersions();
+        
+        ChangePrefixLocationCommand = new AsyncRelayCommand(ChangePrefixLocation);
+        ChangeRunnerLocationCommand = new AsyncRelayCommand(ChangeRunnerLocation);
+        
+        AddEnvCommand = new RelayCommand(AddEnv);
+        DeleteEnvCommand = new RelayCommand(DeleteEnv);
+    }
+
+    private void DeleteEnv()
+    {
+        // TODO: get the selected item and remove it from the dictionary
+    }
+
+    private void AddEnv()
+    {
+        Config.WineEnv.Add("", "");
+    }
```

Comment:
;P

Reply:
Part of the custom environment table structure like in lutris
![image](https://github.com/stayintarkov/SIT.Manager.Avalonia/assets/47221652/24304037-791b-41bd-8f2a-de4d7596493b)